specs/server.js

```diff
@@ -95,6 +95,38 @@ describe('Test API routes', () => {
   });
 });
 
+describe('Test API failure scenarios', () => {
+  it('returns 400 when productId is not a number', () => {
+    request.get("/api/product/quotes?productId='a'")
+      .expect((res) => {
+        expect(res.status).to.equal(400);
+      })
+      .end((err) => {
+        if (err) throw err;
+      });
+  });
+
+  it('returns 404 when productId is a number less than 1', () => {
+    request.get('/api/product/quotes?productId=0')
+      .expect((res) => {
+        expect(res.status).to.equal(404);
+      })
+      .end((err) => {
+        if (err) throw err;
+      });
+  });
+
+  it('returns 404 when productId is a number more than 100', () => {
+    request.get('/api/product/quotes?productId=101')
+      .expect((res) => {
+        expect(res.status).to.equal(404);
+      })
+      .end((err) => {
+        if (err) throw err;
+      });
+  });
```

Comment:
Since you've made a change to the source, this test should be changed also.

Reply:
Not sure why this should change. This test is not doing any static code assessment of the API, instead I am passing a value which I know does not exist in the DB and expecting my API to handle it appropriately. Please elaborate if you still think any change may improve this particular test, as a result of my API handling 404 differently.

Reply:
Your code does not test numbers outside of 0 and 100 any longer so line 119 should be changed to reflect what your source is actually doing. Also, there are ways to replicate a promise that resolves as an empty array so that would again be more accurate to what you are actually testing.

Reply:
What I'm trying to test here, is make a request that forces API to return 404. If I resolved promise in the test suite to an empty array, that would make my API redundant. Not sure if that is what you meant though. As far as the test itself, I will refine line 119 to reflect actual test.

Reply:
As I've stated before, your code no longer tests a 404 on ID numbers outside of 0 and 100; it tests if your query returns an empty array. I did take a closer look at your code and it would seem that this test is dependent on the production database which coincidentally happens to only have products ID's between 0 and 101. Unless I'm mistaken, I do not see a mock or something of the sort which guarantees this passes every time if your product ID's were to expand from 0 - 200 in your seeding script.

Reply:
Ideally the test should break then, as I am seeding upto 100 products by design, any more and that requirement would change. As such, refactoring test then would make sense. My understanding is we test expected outcome, and not really rewrite code including any logic in the test suites, unless absolutely required. And yes, I am not using any mock DB, my understanding is that was not an explicit scope of FEC.

Reply:
In this case, you are not testing the route, but if your database does not seed product IDs outside of 0 and 100. While the tests work, this is not an accurate representation of what you are saying you are testing.

Reply:
I'm confused. Seems to me you have a specific view on how I should be doing this, and I'm not seeing it. Like I mentioned on slack this needs a broader discussion, let's talk when we meet next.